src/main/java/com/mathworks/ci/utilities/MatlabCommandRunner.java

```diff
@@ -0,0 +1,215 @@
+package com.mathworks.ci.utilities;
+
+/**
+ * Copyright 2024, The MathWorks Inc.
+ *
+ */
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.ByteArrayOutputStream;
+import java.util.Map;
+import java.util.HashMap;
+import org.apache.commons.lang.RandomStringUtils;
+
+import hudson.FilePath;
+import hudson.EnvVars;
+import hudson.Launcher;
+import hudson.Launcher.ProcStarter;
+import hudson.model.Computer;
+import hudson.util.ArgumentListBuilder;
+
+import com.mathworks.ci.Utilities;
+import com.mathworks.ci.MatlabExecutionException;
+import com.mathworks.ci.MatlabBuilderConstants;
+import com.mathworks.ci.parameters.MatlabActionParameters;
+
+public class MatlabCommandRunner {
+    private MatlabActionParameters params;
+    private FilePath tempFolder;
+    private OutputStream stdOut;
+    private Map<String,String> additionalEnvVars;
+
+    public MatlabCommandRunner(MatlabActionParameters params) throws IOException, InterruptedException {
+        this.params = params;
+        this.additionalEnvVars = new HashMap<String,String>();
+
+        // Create MATLAB folder
+        FilePath matlabFolder = new FilePath(
+                params.getLauncher().getChannel(), params.getWorkspace().getRemote() 
+                + File.separator 
+                + ".matlab");
+        matlabFolder.mkdirs();
+
+        // Create temp folder
+        this.tempFolder = matlabFolder.createTempDir("tempDir", null);
+        
+        // If we hit an error during shutdown while cleaning up
+        // there's not too much that we can do.
+        Runtime.getRuntime().addShutdownHook(
+                new Thread(() -> {
+                    try {
+                        tempFolder.deleteRecursive();
+                    } catch(Exception e) {
+                        System.err.println(e.toString());
+                    }
+                }));
+    }
+
+    /** 
+     * Spawns a process to run the specified command.
+     *
+     * @param command The command to run
+     */
+    public void runMatlabCommand(String command) throws IOException, InterruptedException, MatlabExecutionException {
+
+        this.params.getTaskListener().getLogger()
+            .println("\n#################### Starting command output ####################");
+
+        // Prepare the executable
+        FilePath exePath = prepareRunnerExecutable();
+
+        // Create the script file
+        FilePath scriptFile = createFileWithContent(command);
+        String cmd = "setenv('MW_ORIG_WORKING_FOLDER', cd('"
+            + this.tempFolder.getRemote()
+            + "'));"
+            + scriptFile.getBaseName();
+
```

Comment:
Oh I see, this fixes the `-sd` thing here, but we need `addPath` only for `genscript` and plugins. Disregard the above commend 👍